implementations/rust/ockam/ockam_api/src/ui/terminal/notification.rs

```diff
@@ -9,12 +9,52 @@ use tokio::select;
 use tokio::sync::broadcast::Receiver;
 use tokio::time::sleep;
 
-use tracing::info;
-
 const REPORTING_CHANNEL_POLL_DELAY: Duration = Duration::from_millis(100);
-const REPORTING_CHANNEL_MESSAGE_DISPLAY_DELAY: Duration = Duration::from_millis(1_000);
 
-pub type Notification = String;
+#[derive(Debug, Clone)]
+pub struct Notification {
```

Comment:
I think it is better to model this as an enum from the start:
```rust
pub enum Notification {
   Message(String),
   ProgressSet(String),
   ProgressFinish(Option<String>),
}
```
This is more precise and then we don't have to worry down the road if the "content" exists or not, we can just pattern match on the 3 cases.

Reply:
done!